.gitlab/docker/Dockerfile.spack_base

```diff
@@ -0,0 +1,47 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG SPACK_REPO=https://github.com/spack/spack
+ARG SPACK_VERSION
```

Comment:
This applies to all dockerfiles: could you please move the ARGs close to where they are actually used? Since every ARG creates a new layer, changing an ARG invalidates all later layers even if an ARG isn't necessarily used by the immediate next RUN commands.